### ISSUE TITLE
Error opening a CSV file with single quote in first line

### ISSUE THREAD

**Issue by AbdealiJK** [jupyterlab/jupyterlab#3189]
I am getting the following error in the javascript console when I open a CSV file:

```
Uncaught (in promise) SyntaxError: Unexpected identifier
    at new Function (<anonymous>)
    at makeRowFn (widget.js:202)
    at widget.js:175
    at Object.parseRows (dsv.js:91)
    at Object.parse (widget.js:171)
    at CSVViewer._updateGrid (widget.js:130)
    at widget.js:65
    at <anonymous>
```

Using the latest jupyterlab master at 212ab379edb0305e14e795c266862c4ba6755ce7
On Chrome 61.0.3163.100 on Mac OSX Sierra

My data file is:
```
1,abc'd
1,2
```

EDIT: After more debugging looks like this happens only when there is a single quote `'` in the first line of the CSV file.

**Comment by blink1073** [comment]
This [line](https://github.com/jupyterlab/jupyterlab/blob/0e51c31d42a948488277d1f64525dfb156e4cc68/packages/csvviewer/src/widget.ts#L247) needs to be updated to replace the `'` character with something else.

**Comment by jasongrout** [comment]
It looks like the substitution in https://github.com/jupyterlab/jupyterlab/blob/0e51c31d42a948488277d1f64525dfb156e4cc68/packages/csvviewer/src/widget.ts#L246 should escape any single quotes in the `${name}` substitution.

**Comment by jasongrout** [comment]
Assigning to 1.0 to give a new contributor a chance to tackle this.

**Comment by apsknight** [comment]
@jasongrout I am new to open-source. Can I work on this issue ?

**Comment by blink1073** [comment]
Yes, please do!

**Comment by blink1073** [comment]
Fixed by #3995

**Comment by Jeroenvanderholst** [comment]
You can't just leav out the single quotes. That is not a solution. I many languages like Italian and French you use single quotes quite often. So therefore you should look for a solution that is importing lines of text with single quotes in after all.
Unfortunately I am too un experienced to give you that solution :(
